### PR TITLE
Dale Worley's Gen-ART comments

### PR DESCRIPTION
**Pull request by ekr**

### CODE REVIEW

File: draft-ietf-tls-tls13.md [hunk 1]
```diff
@@ -3414,18 +3418,18 @@ each included handshake message, including the handshake
...
-     Transcript-Hash(M1, M2, ... MN) = Hash(M1 || M2 ... MN)
+     Transcript-Hash(M1, M2, ... Mn) = Hash(M1 || M2 || ... || Mn)
 
 As an exception to this general rule, when the server responds to a
 ClientHello with a HelloRetryRequest, the value of ClientHello1 is
 replaced with a special synthetic handshake message of handshake
 type "message_hash" containing Hash(ClientHello1). I.e.,
 
-     Transcript-Hash(ClientHello1, HelloRetryRequest, ... MN) =
+     Transcript-Hash(ClientHello1, HelloRetryRequest, ... Mn =
```
Comment (seanturner) [inline draft-ietf-tls-tls13.md]: Is this missing a closing ) ?

Comment (ekr) [reply draft-ietf-tls-tls13.md]: Fixed.

File: draft-ietf-tls-tls13.md [hunk 2]
```diff
@@ -1356,7 +1356,7 @@ described in the following sections.
 
 Although TLS PSKs can be established out of band,
 PSKs can also be established in a previous connection and
-then used to establish a new session ("session resumption"). Once a handshake has completed, the server can
+then used to establish a new connection ("session resumption" or "resuming" with a PSK).
 send to the client a PSK identity that corresponds to a unique key derived from
```
Comment (kaduk) [inline draft-ietf-tls-tls13.md]: We lost the "Once a handshake has completed, the server can"?

File: draft-ietf-tls-tls13.md [hunk 3]
```diff
@@ -2913,9 +2916,9 @@ peers MUST validate each other's public value Y by ensuring
 that the point is a valid point on the elliptic curve.
 The appropriate validation procedures are defined in Section 4.3.7 of {{X962}}
 and alternatively in Section 5.6.2.3 of {{?KEYAGREEMENT=DOI.10.6028/NIST.SP.800-56Ar2}}.
-This process consists of three steps: (1) verify that Y is not the point at
-infinity (O), (2) verify that for Y = (x, y) both integers are in the correct
-interval, (3) ensure that (x, y) is a correct solution to the elliptic curve equation.
+This process consists of three steps: (1) verify that the point (X, Y) is not the point at
+infinity (O), (2) verify that for both X and Y integers are in the correct
```
Comment (kaduk) [inline draft-ietf-tls-tls13.md]: This is a little rough, maybe "verify that both X and Y are integers in the correct interval"?

File: draft-ietf-tls-tls13.md [hunk 4]
```diff
@@ -3598,7 +3602,7 @@ The following rules apply to the certificates sent by the server:
   extension, clients SHOULD send this extension, when applicable.
 
 All certificates provided by the server MUST be signed by a
-signature algorithm advertised by the client, if they are able to provide such
+signature algorithm advertised by the client, if it are able to provide such
```
Comment (kaduk) [inline draft-ietf-tls-tls13.md]: "is is"?

File: draft-ietf-tls-tls13.md [hunk 5]
```diff
@@ -4223,12 +4227,13 @@ by an encrypted body, which itself contains a type and optional padding.
            opaque encrypted_record[TLSCiphertext.length];
        } TLSCiphertext;
 
-content
-: The byte encoding of a handshake or an alert message, or the raw bytes of
-  the application's data to send.
+fragment.
```
Comment (kaduk) [inline draft-ietf-tls-tls13.md]: 'content' was correct. (and no period after 'fragment')

File: draft-ietf-tls-tls13.md [hunk 6]
```diff
@@ -4419,7 +4424,9 @@ specified by the current connection state.
 
 Alert messages convey a description of the alert and a legacy field
 that conveyed the severity of the message in previous versions of
-TLS.  In TLS 1.3, the severity is implicit in the type of alert
+TLS. I think at this point you want to insert "Alerts are divided into
+two classes: closure alerts and error alerts.  In TLS 1.3, the
```
Comment (kaduk) [inline draft-ietf-tls-tls13.md]: "I think at this point"?

File: draft-ietf-tls-tls13.md [hunk 7]
```diff
@@ -4430,13 +4437,14 @@ connection (see {{error-alerts}}). Upon receiving an error alert,
 the TLS implementation SHOULD indicate an error to the application and
 MUST NOT allow any further data to be sent or received on the
 connection.  Servers and clients MUST forget keys and secrets
-associated with a failed connection. Stateful implementations of
-tickets (as in many clients) SHOULD discard tickets associated
+used with failed connection. Stateful implementations of
```
Comment (kaduk) [inline draft-ietf-tls-tls13.md]: "used with failed connection" doesn't parse.

File: draft-ietf-tls-tls13.md [hunk 8]
```diff
@@ -5090,7 +5104,9 @@ by {{pre-shared-key-extension}}.
 I.e., if the
 client sends PSKs A and B but the server prefers A, then the
 attacker can change the binder for B without affecting the binder
-for A. This will cause the ClientHello to be accepted, and may
+for A. If the binder for B is part of the storage key,
+then this ClientHello, will not appear as a duplicate,
```
Comment (kaduk) [inline draft-ietf-tls-tls13.md]: No comma before "will"

File: draft-ietf-tls-tls13.md [hunk 9]
```diff
@@ -5165,8 +5181,9 @@ handshake can be allowed to complete.
 There are several potential sources of error that might cause
 mismatches between the expected arrival time and the measured
 time. Variations in client and server clock
-rates are likely to be minimal, though potentially with gross time
-corrections.  Network propagation delays are the most likely causes of
+rates are likely to be minimal, though potentially the absolute
+times may of off by large values.
```
Comment (kaduk) [inline draft-ietf-tls-tls13.md]: s/may of/may be/

**Comment by ekr** [comment]
Not my best day.

File: draft-ietf-tls-tls13.md [hunk 10]
```diff
@@ -4436,10 +4437,10 @@ Error alerts indicate abortive closure of the
 connection (see {{error-alerts}}). Upon receiving an error alert,
 the TLS implementation SHOULD indicate an error to the application and
 MUST NOT allow any further data to be sent or received on the
-connection.  Servers and clients MUST forget keys and secrets
-used with failed connection. Stateful implementations of
-tickets (as in many clients) SHOULD discard tickets established on
-with failed connections.
+connection.  Servers and clients MUST the secret values and
```
Comment (kaduk) [inline draft-ietf-tls-tls13.md]: We lost "forget", it looks like?

**Comment by ekr** [comment]
@kaduk: PTAL now

**Comment by ekr** [comment]
Merged

**Pull request closed without merging**